File: ios/CountryData.m
//
//  CountryData.m
//  PrestaSign
//
//  Created by Morgan Mutugi on 17/04/2023.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(CountryData, NSObject)

RCT_EXTERN_METHOD(getCountries:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject);

@end

File: ios/CountryData.swift
//
//  CountryData.swift
//  PrestaSign
//
//  Created by Morgan Mutugi on 17/04/2023.
//

import Foundation

@objc(CountryData)
class CountryData: NSObject {
  
  @objc func constantsToExport() -> [String: Any] {
    return [
      "name": "CountryData",
    ]
  }
  
  @objc
  static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc
  func getCountries(_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
    if let url = Bundle.main.url(forResource: "Countries", withExtension: "json") {
      do {
        let data = try Data(contentsOf: url, options: .mappedIfSafe)
        let jsonObject = try JSONSerialization.jsonObject(with: data, options: .mutableLeaves)
        resolve(jsonObject)
      } catch {
        reject("getCountries", "Could not serialize country data", nil)
      }
    } else {
      reject("getCountries", "Url to countries file not available", nil)
    }
  }
}
